src/Castle.Core/DynamicProxy/ModuleScope.cs

```diff
@@ -565,7 +657,7 @@ public void LoadAssemblyIntoCache(Assembly assembly)
 
 				if (loadedType != null)
 				{
-					RegisterInCache(mapping.Key, loadedType);
+					typeCache[mapping.Key] = loadedType;
```

Comment:
Note the absence of any locking here. Proper locking should be added, I didn't do this yet because I wanted this PR to be mostly a refactoring.